cdi/embedded/src/main/java/org/infinispan/cdi/InfinispanExtensionEmbedded.java

```diff
@@ -72,18 +78,28 @@ void processProducers(@Observes ProcessProducer<?, ?> event, BeanManager beanMan
    // This is a work around for CDI Uber Jar deployment. When Weld scans the classpath it  pick up DefaultCacheManager
    // (this is an implementation, not an interface, so it gets instantiated). As a result we get duplicated classes
    // in CDI BeanManager.
-   @SuppressWarnings("unused")
    <T extends DefaultCacheManager> void removeDuplicatedRemoteCacheManager(@Observes ProcessAnnotatedType<T> bean) {
        if(DefaultCacheManager.class.getCanonicalName().equals(bean.getAnnotatedType().getJavaClass().getCanonicalName())) {
-          logger.info("removing duplicated  DefaultCacheManager" + bean.getAnnotatedType());
+          LOGGER.info("removing duplicated  DefaultCacheManager" + bean.getAnnotatedType());
           bean.veto();
        }
    }
 
    @SuppressWarnings("unchecked")
-   <T, X>void registerCacheBeans(@Observes AfterBeanDiscovery event, final BeanManager beanManager) {
+   <T, X>void registerBeans(@Observes AfterBeanDiscovery event, final BeanManager beanManager) {
+
+      if (beanManager.getBeans(Configuration.class).isEmpty()) {
+         LOGGER.addDefaultEmbeddedConfiguration();
+         // TODO simulate processProducers()/@ConfigureCache
```

Comment:
I believe we should do it. Looking at [the code](https://github.com/infinispan/infinispan/blob/e1930ec92a3191008cdb248ebec1ccebbb1d82c4/cdi/embedded/src/main/java/org/infinispan/cdi/InfinispanExtensionEmbedded.java#L63-L69) I see that we register `AdvancedCache`es (and `Cache`es) only for configuration annotated with `ConfigureCache`. This will allow users to create their own producers for `Configuration` which won't be touched by Infinispan extension. This approach makes sense to me. And of course it needs to be mentioned in the docs (but no worries I'll handle this).